### PR TITLE
Integration/portable migration

### PR DESCRIPTION
**Pull request by ssipasseuth**
Reimplemented the life cycle of PCI according to https://oat-sa.atlassian.net/browse/TAO-2524

Requires https://github.com/oat-sa/extension-tao-itemqti/pull/765
Please check this related PR for more information.

### CODE REVIEW

File: test/samples/package/likertScaleInteraction_v1.0.0/creator/widget/Widget.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+define([
```
Comment (jsconan) [inline test/samples/package/likertScaleInteraction_v1.0.0/creator/widget/Widget.js]: Missing header

File: test/samples/package/likertScaleInteraction_v1.0.0/creator/widget/Widget.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+    'taoQtiItem/qtiCreator/widgets/interactions/customInteraction/Widget',
+    'likertScaleInteraction/creator/widget/states/states'
+], function(Widget, states){
+
```
Comment (jsconan) [inline test/samples/package/likertScaleInteraction_v1.0.0/creator/widget/Widget.js]: Missing "use strict"

File: test/samples/package/likertScaleInteraction_v1.0.0/creator/widget/states/Question.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
+define([
```
Comment (jsconan) [inline test/samples/package/likertScaleInteraction_v1.0.0/creator/widget/states/Question.js]: Missing header

File: test/samples/package/likertScaleInteraction_v1.0.0/creator/widget/states/Question.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+    'taoQtiItem/qtiCreator/widgets/states/factory',
+    'taoQtiItem/qtiCreator/widgets/interactions/states/Question',
+    'taoQtiItem/qtiCreator/widgets/helpers/formElement',
+    'taoQtiItem/qtiCreator/editor/simpleContentEditableElement',
+    'taoQtiItem/qtiCreator/editor/containerEditor',
+    'tpl!likertScaleInteraction/creator/tpl/propertiesForm',
+    'lodash',
+    'jquery'
+], function(stateFactory, Question, formElement, simpleEditor, containerEditor, formTpl, _, $){
+
```
Comment (jsconan) [inline test/samples/package/likertScaleInteraction_v1.0.0/creator/widget/states/Question.js]: Missing "use strict"

File: views/js/pciProvider.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    var _serverUrl = helpers._url('load', 'PciLoader', 'qtiItemPci');
+    
+    return {
+        load: function load(){
+            return new Promise(function(resolve){
+                $.ajax({
+                    url : _serverUrl,
+                    dataType : 'json',
+                    type : 'GET'
+                }).done(function (pcis){
```
Comment (jsconan) [inline views/js/pciProvider.js]: What happens if the query fails ?

File: views/js/pciManager/pciManager.js [hunk 6]
```diff
@@ -7,14 +25,15 @@ define([
...
+    'taoQtiItem/portableElementRegistry/ciRegistry',
     'async',
+    'ui/dialog/confirm',
     'ui/deleter',
     'ui/feedback',
     'ui/modal',
     'ui/uploader',
     'ui/filesender'
-], function($, __, _, helpers, layoutTpl, listingTpl, packageMetaTpl, interactionsToolbar, ciRegistry, async, deleter, feedback){
+], function($, __, _, helpers, layoutTpl, listingTpl, packageMetaTpl, interactionsToolbar, ciRegistry, async, confirmBox, deleter, feedback){
```
Comment (jsconan) [inline views/js/pciManager/pciManager.js]: Missing "use strict";

**Comment by ssipasseuth** [comment]
I have fixed the issues, please take a look again.

**Comment by siwane** [comment]
This PR will be replaced by branch https://github.com/oat-sa/extension-tao-itemqti-pci/tree/feature/portableElement

**Pull request closed without merging**